contracts/FungibleTokenMetadataViews.cdc

```diff
@@ -0,0 +1,177 @@
+
+        /// Banner-sized image for this collection, recommended to have a size near 1200x630.
+        pub let bannerImage: MetadataViews.Media
+
+        /// Social links to reach this collection's social homepages.
+        /// Possible keys may be "instagram", "twitter", "discord", etc.
+        pub let socials: {String: MetadataViews.ExternalURL}
+
+        init(
+            name: String,
+            description: String,
+            thumbnail: AnyStruct{MetadataViews.File},
+            externalURL: MetadataViews.ExternalURL,
+            squareImage: MetadataViews.Media,
+            bannerImage: MetadataViews.Media,
+            socials: {String: MetadataViews.ExternalURL}
+        ) {
+            self.name = name
+            self.description = description
+            self.thumbnail = thumbnail
+            self.externalURL = externalURL
+            self.squareImage = squareImage
+            self.bannerImage = bannerImage
+            self.socials = socials
+        }
+    }
+
+    /// Helper to get FTDisplay in a way that will return a typed optional
+    /// 
+    /// @param viewResolver: A reference to the resolver resource
+    /// @return A optional FTDisplay struct
+    ///
+    pub fun getFTDisplay(_ viewResolver: &{MetadataViews.Resolver}): FTDisplay? {
+        if let view = viewResolver.resolveView(Type<FTDisplay>()) {
+            if let v = view as? FTDisplay {
+                return v
+            }
+        }
+        return nil
+    }
+
+    /// View to expose the information needed store and interact with a FT vault.
+    /// This can be used by applications to setup a FT vault with proper 
+    /// storage and public capabilities.
+    ///
+    pub struct FTVaultData {
+        /// Path in storage where this FT vault is recommended to be stored.
+        pub let storagePath: StoragePath
+
+        /// Public path which must be linked to expose public capabilities of
+        /// this FT, including standard FT interfaces and metadataviews 
+        /// interfaces
+        pub let publicPath: PublicPath
+
+        /// Private path which should be linked to expose the provider
+        /// capability to withdraw funds from the vault
+        pub let providerPath: PrivatePath
+
+        /// Type that should be linked at the aforementioned public path. This 
+        /// is normally a restricted type with many interfaces. Notably the 
+        /// `FungibleToken.Balance`, `FungibleToken.Receiver`, and  
+        /// `MetadataViews.Resolver` interfaces are required.
+        pub let publicLinkedType: Type
+
+        /// Type that should be linked at the aforementioned private path. This 
+        /// is normally a restricted type with at a minimum the  
+        /// `FungibleToken.Provider` interface
+        pub let providerLinkedType: Type
+
+        /// Function that allows creation of an empty FT vault that is intended
+        /// to store the funds.
+        pub let createEmptyVault: ((): @FungibleToken.Vault)
```

Comment:
What is the reasoning for this `createEmptyVault` function to be inside the `FTVaultData` struct?

Reply:
People can get metadata views from an object without importing or knowing what contract the object comes from, so this view allows code that has the object or a reference to it, but not the imported contract, to be able to create a new vault and also know what the storage and public paths are for it without having to find out that information elsewhere.